src/pages/iou/steps/IOUAmountPage.js

```diff
@@ -228,26 +218,16 @@ class IOUAmountPage extends React.Component {
                     styles.justifyContentCenter,
                 ]}
                 >
-                    <TouchableOpacity onPress={() => Navigation.navigate(this.props.hasMultipleParticipants
-                        ? ROUTES.getIouBillCurrencyRoute(this.props.reportID)
-                        : ROUTES.getIouRequestCurrencyRoute(this.props.reportID))}
-                    >
-                        <Text style={styles.iouAmountText}>
-                            {lodashGet(this.props.currencyList, [this.props.iou.selectedCurrencyCode, 'symbol'])}
-                        </Text>
-                    </TouchableOpacity>
-                    <TextInput
-                        disableKeyboard
-                        autoGrow
-                        hideFocusedState
-                        inputStyle={[styles.iouAmountTextInput, styles.p0, styles.noLeftBorderRadius, styles.noRightBorderRadius]}
-                        textInputContainerStyles={[styles.borderNone, styles.noLeftBorderRadius, styles.noRightBorderRadius]}
-                        onChangeText={this.updateAmount}
-                        ref={el => this.textInput = el}
-                        value={formattedAmount}
+                    <TextInputWithCurrencySymbol
+                        currencySymbol={currencySymbol}
+                        formattedAmount={formattedAmount}
+                        isCurrencySymbolLTR={isCurrencySymbolLTR}
+                        onChangeAmount={this.updateAmount}
+                        onCurrencyButtonPress={() => Navigation.navigate(this.props.hasMultipleParticipants
+                            ? ROUTES.getIouBillCurrencyRoute(this.props.reportID)
+                            : ROUTES.getIouRequestCurrencyRoute(this.props.reportID))}
```

Comment:
I think we can create a class method and use it here. It will also be better from the performance point of view.